engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1359,7 +1409,8 @@ private void addDAGToCurrentDAG(ModuleMeta moduleMeta)
     String name;
     for (OperatorMeta operatorMeta : subDag.getAllOperators()) {
       name = subDAGName + MODULE_NAMESPACE_SEPARATOR + operatorMeta.getName();
-      this.addOperator(name, operatorMeta.getOperator());
+      Operator op = this.addOperator(name, operatorMeta.getOperator());
```

Comment:
We are adding an operator with a different name to a different DAG and as result the new meta object is created. Naturally that's a new object vs. sharing the meta object between DAGs... The meta object is result of the add operation and the name field should remain final. I don't see the need to add a copy constructor, setting attributes is public API.